source/docs/ondemand_rds_vhds.md

```diff
@@ -0,0 +1,161 @@
+
+## Unsubscribing from Virtual Hosts
+Virtual hosts can be unsubscribed to via a [DeltaDiscoveryRequest](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L137) with their route config names + domains provided in the resource_names_unsubscribe field. Envoy will remove any route config names + domains that it finds in the [DeltaDiscoveryRequest](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L137) removed_resources field. 
+
+## Subscribing to Virtual Hosts
+Envoy will send a [DeltaDiscoveryRequest](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L137) with the resource_names_subscribe field populated with the route config names + domains of each of the resources that it would like to subscribe to. Each of the virtual hosts contained in the [DeltaDiscoveryRequest's](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L137) resources field will be added to the route configuration maintained by Envoy. If Envoy's route configuration already contains a given virtual host, it will be overwritten by data received in the [DeltaDiscoveryResponse](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L193), but only if the updated virtual host is different from its current state. During spontaneous updates configuration server will only send updates for virtual hosts that Envoy is aware of -- the configuration server needs to keep track of virtual hosts known to Envoy.
+
+## Updates of non-virtual host fields
+It might be required to update non-virtual host fields of route configuration while keeping virtual hosts as is. This would probably happen during spontaneous updates only. In such a case a [DeltaDiscoveryResponse](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L193) containing a route configuration with an empty virtual hosts list can be used.
+
+## Compatibility with Scoped RDS
+Both approaches appear to be compatible with scoped RDS proposal: route_configuration name can still be used for vhost matching, but with scoped RDS configured it would point to a scoped route configuration.
+
+Pilot will also support the ability of Envoy to tell it when a resource hasn’t been used and is safe to stop monitoring. The resources that can be removed include the base resources that Pilot initially sent Envoy. 
```

Comment:
Please replace all references to Pilot with "xDS management server"

Reply:
Thought I'd replaced all of the Pilot references to config server. Let me fix that. Should I change the config server references to xDS management server as well?

Reply:
Yep, having consistent terminology here would be great. We tend to use "xDS management server" in other parts of Envoy.

Reply:
Done. Thanks